echo-telemetry/src/main/java/com/netflix/spinnaker/echo/telemetry/TelemetryEventListener.java

```diff
@@ -217,4 +216,61 @@ public String toString() {
       return new String(getBytes(), StandardCharsets.UTF_8);
     }
   }
+
+  // Arbitrary container for TelemetryEventListener structured data, so there aren't huge
+  // fully-qualified names for the equivalent classes in the kork.proto package all over the code.
+  public static class TEL {
+
+    @Data
```

Comment:
It's a bit more annoying to write these in a way that can be easily deserialized, but I've been trying write as many of my data classes as possible to be immutable.  In this case I don't think we're ever mutating instances of these classes after creation.

Reply:
Copied the pattern you pointed me to from Orca - I think this ended up with more code rather than the previous null-checking. I think it would have more impact if I were able to actually put this in the `model` package and use it all over the code, rather than just in this local file.